deep-learning/_posts/2018-11-12-AutomaticallyDesigningCNNsforSegmentation.md

```diff
@@ -0,0 +1,114 @@
+---
+layout: review
+title: Automatically Designing CNN Architectures for Medical Image Segmentation
+tags: deep-learning CNN segmentation medical reinforcement cardiac
+cite:
+    authors: "Aliasghar Mortazi and Ulas Bagci"
+    title:   "Automatically Designing CNN Architectures for Medical Image Segmentation"
+    venue:   "MLMI 2018"
+pdf: "https://arxiv.org/pdf/1807.07663.pdf"
+---
+
+
+# Introduction
+
+The authors of this paper propose a new and simple technique to automatically design neural networks using policy gradients.
+The technique involves learning the optimal hyperparameters for an encoder-decoder network used for segmentation with the ACDC dataset. An optimal policy is found by iteratively permutating a random policy and updating it with the permutations that yielded the best reward (i.e. dice index).
+
+
+
+# Method
+![](/deep-learning/images/AutomaticallyDesigningCNN/fig1.jpg)
+
+In this method, each hyperparameter is considered a policy.
+The method starts with a base policy, initialized randomly.
+
+Assume this random policy as $$\pi_0 = \{\theta_1, \theta_2, ..., \theta_N\} $$ indicating the hyperparamters of the network. N is the number of hyperparameters.
+
+New policies are generated by applying random permutations to the policy in each dimension (for each hyperparameter)
+
+Let:
+
+$$
+P(\pi_0) = \{\pi_1, \pi_2,..., \pi_p\}
+$$
+
+define p random pemutations near $$\pi_0$$, where:
+
+$$
+\pi_i = \pi_0 + \Delta_i \text{ for } i \in \{1, 2,..., p\}
+$$
+
+For each permutation $$\pi_i$$, $$\Delta_i$$ is defined as:
+
+$$
+\Delta_i = \{\delta^1, \delta^2,..., \delta^N\}
+$$
+
+where $$\delta^d$$ is a randomly chosen from $$\{-\epsilon^d, 0, +\epsilon^d\}$$ with $$d \in \{1, 2,..., N\}$$. $$\epsilon$$ is the derivative of a function $$y$$ with respects to $$x$$.
+These functions are the learnable hyperparameters for each layer:
+
+* Number of filters: $$y_{NF} = 16x_{NF} + 16 \text{ with } x_{NF} = \{1, 2,..., 12\}$$
+* Filter height: $$y_{FH} = 2x_{FH} + 1 \text{ with } x_{FH} = \{0, 1,..., 5\}$$
+* Filter width: $$y_{FW} = 2x_{FW} + 1 \text{ with } x_{FW} = \{0, 1,..., 5\}$$
+* Pooling function: $$y_{pooling} = x_{pooling} \text{ with } x_{pooling} = \{0, 1\}$$ where 0 is maxpooling and 1 is average pooling.
+
+The segmentation network is trained with the p generated policies and a reward is obtained from each policy.
+
+To estimate the partial derivate of the policy function for each dimension (hyperparameter), each pertuation is grouped to non-overlapping categores of negative pertubations, zero pertubations and positive pertubations. $$C^d_-$$, $$C^d_0$$ and $$C^d_+$$ such that $$\pi^d_i \in \{C^d_-, C^d_0, C^d_+\}$$. This categorizations is based on the permutation that was applied in that dimension ($$\{-\epsilon^d, 0, +\epsilon^d\} $$).
+
```

Comment:
`This categorizations is based (...)`: either `This categorization is based (...)` or `This categorizations are based (...)`